pkg/prometheus/promcfg.go

```diff
@@ -92,11 +92,28 @@ func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, e
 		return nil, fmt.Errorf("failed to parse scrape classes: %w", err)
 	}
 
+	endpointSliceSupportedAndConfigured := false // Always assume false to preserve original prometheus-operator behaviour.
+
+	// Check if the user has explicitly set the service discovery role to use.
+	switch serviceDiscoveryRole := *cpf.ServiceDiscoveryRole; serviceDiscoveryRole {
+	case "EndpointSlice":
```

Comment:
```suggestion
	case monitoringv1.EndpointSliceRole:
```

Reply:
Thank you for the code style suggestion. I have implemented this in [7d5c6a5](https://github.com/prometheus-operator/prometheus-operator/pull/6672/commits/7d5c6a5421819696a553ae02c0af9df0f6b2beb2)